### PR TITLE
Added -ss --step-size option for larger scan point distances

### PR DESCRIPTION
**Pull request by Mantriur**
See https://github.com/PokemonGoMap/PokemonGo-Map/issues/72

 -ss STEP_SIZE, --step-size STEP_SIZE
                        Distance between scan points, 70 is default and max
                        for pokemons, use with -np.

### CODE REVIEW

**Comment by Mantriur** [comment]
Don't merge this yet, apparently the Default option is purlely informational.

**Comment by Mantriur** [comment]
Fairly certain this works properly now.

**Comment by karololszak** [comment]
FYI in `static/js/map.js` there is a `setupScannedMarker` function that draws the scan radius on map and there's a `radius: 70, // metres` param there.

```
function setupScannedMarker (item) {
  var circleCenter = new google.maps.LatLng(item['latitude'], item['longitude'])

  var marker = new google.maps.Circle({
    map: map,
    clickable: false,
    center: circleCenter,
    radius: 70, // metres
    fillColor: getColorByDate(item['last_modified']),
    strokeWeight: 1
  })

  return marker
}
```

It can be changed to reflect the current scannedMarker. I don't know how you'd have to go about it, maybe a param to the function, maybe a global variable? Ask frontend devs :>

File: pogom/search.py [hunk 1]
```diff
@@ -57,16 +58,19 @@ def get_new_coords(init_loc, distance, bearing):
     return [math.degrees(new_lat), math.degrees(new_lon)]
 
 def generate_location_steps(initial_loc, step_count):
+
+    # There's another call to this *not* using args.step_count as second argument, this is for step_size only.
```
Comment (chuyskywalker) [inline pogom/search.py]: > There's another call to this *not* using args.step_count as second argument

What's that got to do with anything? This code comment is confusing...

File: pogom/utils.py [hunk 2]
```diff
@@ -48,6 +48,9 @@ def get_args():
                         help='Location, can be an address or coordinates')
     parser.add_argument('-st', '--step-limit', help='Steps', type=int,
                         default=12)
+    parser.add_argument('-ss', '--step-size',
+                        help='Distance between scan points, 70 is default and max for pokemons, use with -np.',
```
Comment (chuyskywalker) [inline pogom/utils.py]: > Distance between scan points, 70 is default and max for pokemons, use with -np.

> Distance between scan points, useful for scanning gyms/pokestops at wider range. Does not pickup pokemon more effectively.

Or something like that.

File: pogom/search.py (hunk 1 above)
Comment (Mantriur) [inline pogom/search.py]: The functions second step_count argument is fed from args.step_count further down, it looks like an obvious optimization freebie to lose the second argument and use args.step_count instead. However, generate_location_steps() is used in another place in the code, where it is fed a different value.

File: pogom/utils.py (hunk 2 above)
Comment (Mantriur) [inline pogom/utils.py]: Force -ss to 70 unless used with -np? I see no real loss there, it *is* useless for pokemons.

File: pogom/search.py [hunk 3]
```diff
@@ -57,16 +58,24 @@ def get_new_coords(init_loc, distance, bearing):
...
+    args = get_args()
+    
     #Bearing (degrees)
     NORTH = 0
     EAST = 90
     SOUTH = 180
     WEST = 270
 
-    pulse_radius = 0.07                 # km - radius of players heartbeat is 70m
+    if args.no_pokemon: # scan_size requires -np, 70 is default
```
Comment (chuyskywalker) [inline pogom/search.py]: This should never be unset, though, right? The args has a default value...

File: pogom/search.py [hunk 4]
```diff
@@ -57,16 +58,24 @@ def get_new_coords(init_loc, distance, bearing):
     return [math.degrees(new_lat), math.degrees(new_lon)]
 
 def generate_location_steps(initial_loc, step_count):
+
+    # Do not use 'args" to replace the "step_count" parameter above, it does not always equal args.step_count!
```
Comment (chuyskywalker) [inline pogom/search.py]: When else is it called, I only found one usage.

Comment (chuyskywalker) [inline pogom/search.py]: Ahh, in the mock generator -- that's not so important

Comment (Mantriur) [inline pogom/search.py]: It's also in util.py insert_mock_data():

`locations = [l for l in generate_location_steps((latitude, longitude), num_pokemon)]`

**Comment by chuyskywalker** [comment]
@karololszak you would specifically NOT want to modify the client side code. This does not scan a larger area *at all*

**Comment by karololszak** [comment]
@chuyskywalker it **does** scan a larger area, an area of 1km^2, for pokestops and gyms. And changing the client-side code to show bigger circles would fill the gaps in between them.

This shouldn't be a set-variable PR, we should go with "scan for pokemons: 70m; scan for gyms and pokestops: 1000m" -- pokemon scan is of fixed value, so gyms/pokestops scan should be fixed as well. There is no point doing 500m circles, it's a waste of time.

**Comment by Mantriur** [comment]
@karololszak It would require the radius to be added to the database, as it can vary within the same dataset. This is really for a single very specific case and imho not worth the hassle.

**Comment by karololszak** [comment]
If you're thinking of a scenario where one worker is doing the big scan (for the gyms and pokestops) and there are other workers doing the pokemons, leaving it as it is would imply that it scanned for pokemons there as well (it's the same size of circle, same color etc).

Do whatever you want with the frontend, I don't really care about it, I just wanted to point it out.

**Comment by Mantriur** [comment]
Well, yeah, if you'd add color coding, so the user can see the type of scan, it'd be kinda neat to have the correct circle size. Though that's outside the scope of what I had planned here. :/

**Comment by Mantriur** [comment]
Not familiar with github: Is there anything left for me to do? I don't see a way to trigger another Landscape test or to view the conflicts.

**Comment by chuyskywalker** [comment]
I'm pretty sure we just told landscape to take a hike, so don't worry about it.

That said, it looks like you need to rebase a bit at this point.

**Pull request closed without merging**